grr/server/grr_response_server/fleetspeak_utils_test.py

```diff
@@ -133,6 +133,35 @@ def testRestartFleetspeakGrrService(self, mock_conn):
     fs_message.data.Unpack(restart_req)
     self.assertEqual(restart_req.name, "GRR")
 
+  def testFetchClientResourceUsageRecordsFromFleetspeak(self):
+    conn = mock.MagicMock()
+    conn.outgoing.FetchClientResourceUsageRecords.return_value = admin_pb2.FetchClientResourceUsageRecordsResponse(
+        records=[{
+            "mean_user_cpu_rate": 1,
+            "max_system_cpu_rate": 2
+        }, {
+            "mean_user_cpu_rate": 4,
+            "max_system_cpu_rate": 8
+        }])
+    with mock.patch.object(fleetspeak_connector, "CONN", conn):
+      expected_records_list = [
+          resource_pb2.ClientResourceUsageRecord(mean_user_cpu_rate=1,
+                                                 max_system_cpu_rate=2),
+          resource_pb2.ClientResourceUsageRecord(mean_user_cpu_rate=4,
+                                                 max_system_cpu_rate=8)
+      ]
+      self.assertListEqual(
+          fleetspeak_utils.FetchClientResourceUsageRecordsFromFleetspeak(
+              _TEST_CLIENT_ID, 10), expected_records_list)
+      conn.outgoing.FetchClientResourceUsageRecords.assert_called_once()
+      insert_args, _ = conn.outgoing.FetchClientResourceUsageRecords.call_args
```

Comment:
"insert_args" sounds as an odd name. Is it a copy-paste artifact? Also, would it make sense to use https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with instead of getting the call_args and then doing assertEqual?

Reply:
It isn't an artifact, but it is implemented in a similar way in existing code in tests for other `fleetspeak_utils.py` functions. For instance:
https://github.com/google/grr/blob/b472626b6ff4d05b24269a7ce52419d00a847607/grr/server/grr_response_server/fleetspeak_utils_test.py#L112-L117
However I do agree with you that using `assert_called_with` is far more readable. In this case, you actually do not even need to use `assertEqual`.